src/components/Weather/TodayWeatherIcon.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react';
+import css from './TodayWeatherIcon.module.scss';
+
+const TodayWeatherIcon = ({
+  weatherData,
+  todayTime,
+}: {
+  weatherData: any;
+  todayTime: any;
+}) => {
+  const [weatherIcon, setWeatherIcon] = useState<string>();
+  const time = Number(todayTime);
+
+  useEffect(() => {
+    if (time >= 18 || (time >= 0 && time < 5)) {
+      if (weatherData === 'Clear') {
+        setWeatherIcon('./image/night/clear sky.png');
+      } else if (weatherData === 'Clouds') {
+        setWeatherIcon('./image/night/few clouds.png');
+      } else if (weatherData === 'Drizzle') {
+        setWeatherIcon('./image/night/shower rain.png');
+      } else if (weatherData === 'Rain') {
+        setWeatherIcon('./image/night/rain.png');
+      } else if (weatherData === 'Thunderstorm') {
+        setWeatherIcon('./image/night/thunderstorm.png');
+      } else if (weatherData === 'Snow') {
+        setWeatherIcon('./image/night/snow.png');
+      } else {
+        setWeatherIcon('./image/night/mist.png');
+      }
+    } else {
+      if (weatherData === 'Clear') {
+        setWeatherIcon('./image/day/clear sky.png');
+      } else if (weatherData === 'Clouds') {
+        setWeatherIcon('./image/day/few clouds.png');
+      } else if (weatherData === 'Drizzle') {
+        setWeatherIcon('./image/day/shower rain.png');
+      } else if (weatherData === 'Rain') {
+        setWeatherIcon('./image/day/rain.png');
+      } else if (weatherData === 'Thunderstorm') {
+        setWeatherIcon('./image/day/thunderstorm.png');
+      } else if (weatherData === 'Snow') {
+        setWeatherIcon('./image/day/snow.png');
+      } else {
+        setWeatherIcon('./image/day/mist.png');
+      }
+    }
+  }, [weatherData, todayTime]);
```

Comment:
Map을 써서 좀 더 보기 좋게 해볼 수 있을 것 같아서 대충 예시 해봤어, ㅎ.ㅎ

```suggestion
const wetherMap: Map<string, string[]> = new Map([
  ['Clear', ['./image/night/clear sky.png', './image/day/clear sky.png']],
  ['Clouds', ['./image/night/few clouds.png', './image/day/few clouds.png']],
]);

  useEffect(() => {
    const data = wetherMap.get(weatherData);
    const isNight = time >= 18 || (time >= 0 && time < 5);
    if (isNight) {
      if (data) {
        setWeatherIcon(data[0]);
      } else {
        setWeatherIcon('./image/night/mist.png');
      }
    } else {
      if (data) {
        setWeatherIcon(data[1]);
      } else {
        setWeatherIcon('./image/day/mist.png');
      }
    }
  }, [weatherData, todayTime]);
```

Reply:
왕 훨씬 보기좋다,,,진짜루,, 코드 수정해볼게!!!

Reply:
수정완료 😄